packages/grafana-ui/src/components/DateTimePickers/DateTimePicker/DateTimePicker.tsx

```diff
@@ -207,8 +207,9 @@ const DateTimeInput = React.forwardRef<HTMLInputElement, InputProps>(
     }, []);
 
     const onBlur = useCallback(() => {
-      if (isDateTime(internalDate.value)) {
-        onChange(dateTime(internalDate.value));
+      const date = dateTime(internalDate.value);
+      if (isDateTime(date)) {
```

Comment:
this no longer needs wrapping in the if, since calling the function beforehand guarantees it's a `DateTime`